Jenkinsfile

```diff
@@ -0,0 +1,71 @@
+node ('virtualbox') {
+
+  def directory = "packer-template"
+  env.PACKER_CACHE_DIR = "${env.JENKINS_HOME}/packer_cache"
+  stage 'Clean up'
+  deleteDir()
+
+  stage 'Checkout'
+  sh "mkdir $directory"
+  dir("$directory") {
+    try {
+        checkout scm
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    }
+  }
+  dir("$directory") {
+    stage 'bundle'
+    try {
+        sh "bundle install --path ${env.JENKINS_HOME}/vendor/bundle"
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    }
+
+    stage 'rake reallyenglish:build'
+    try {
+      sh 'bundle exec rake reallyenglish:test'
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    } finally {
+      sh 'bundle exec rake reallyenglish:clean'
```

Comment:
here is my understanding: `Jenkinsfile` calls the `reallyenglish:test` task, which calls `do_test` with each box, and `do_test` task fails, say in `build`. `finally` in `Jenkins` file calls `reallyenglish:clean`, which runs `vagrant destroy -f` and `rm *.box`.

would you tell me which part of the process you are talking?